docs/x509.rst

```diff
@@ -196,6 +196,24 @@ X.509 Certificate Object
             >>> isinstance(cert.signature_hash_algorithm, hashes.SHA256)
             True
 
+    .. attribute:: extensions
+
+        :type: :class:`collections.OrderedDict`
```

Comment:
We discussed that a bit in this PR. OrderedDict captures the behavior we want here very well (preserve order, but at most one of each), so I'd like to use that, but I'd be okay with creating an `Extensions` that has a `get_by_oid` to match what we do on `Name` (but also implements `__get_item__`).